### ISSUE TITLE
Lexical binding?

### ISSUE THREAD

**Issue by alphapapa** [facetframer/orgnav#1]
Hi there,

I just found this from https://emacs.stackexchange.com/questions/32617/how-to-jump-directly-to-an-org-headline/32638#32638, which I found from /r/emacs.  I appreciate your mentioning helm-org-rifle in your readme!

This looks like a really cool project!  I will be studying it and perhaps borrowing some ideas for helm-org-rifle.  The way you've structured your code is very interesting.  I have an old branch in helm-org-rifle where I experimented with a more functional approach to searching and building results, but it seemed slower, so I left it behind, so to speak.

I noticed your using `lexical-let` in one of your functions.  I wondered, have you thought about using lexical-binding instead?  As far as I understand, it's considered a best-practice for Emacs packages now, and may help performance as well.  Most of the time it doesn't break anything, either.

Thanks.

**Comment by facetframer** [comment]
Whoops. Sorry for the slow reply. I don't think I have set up my alerts for this account very well. I shall fix this shortly

> I noticed your using lexical-let in one of your functions. I wondered, have you thought about using lexical-binding instead? 

Thanks for the heads-up. This is just me not being knowledgeable about emacs features. 
I'm just reading through the documentation for this now.

Using `lexical-binding` seems reasonable. I think I'm using `defvar` everywhere where I want something to be dynamically overridable by callers.

The only downside with removing `lexical-let`, if my understanding is correct, is that I would break this package of emacs versions younger than 24.1 (https://www.gnu.org/software/emacs/manual/html_node/elisp/Lexical-Binding.html). Looking at https://www.emacswiki.org/emacs/EmacsReleaseDates this seems to be five years old or so. People using org will probably use newish version of emacs.

This seems like a perfectly reasonable change to make.

**Comment by facetframer** [comment]
Okay, I've switched over to lexical-binding and things seem to be mostly working fine. 
I'm going to hold off releasing it until it's sat in my emacs for a while.

I've created a notion of releases (git tags), so that people can have some form of stability. I intend to have quite a low barrier  for me to push to master.

https://github.com/facetframer/orgnav/commit/82e2c76cb49bfb83da149b8dc354e26dbf9f3f03
